### PR TITLE
Robustify updater plugin + pyfakefs + clean code

### PR DESCRIPTION
**Pull request by Wescoeur**
Based on #30

### CODE REVIEW

File: SOURCES/etc/xapi.d/plugins/updater.py [hunk 1]
```diff
@@ -32,30 +32,29 @@ def __init__(self, operation, timeout=0):
         self.operation = operation
 
     def _raise_busy(self):
-        raise OperationException('The updater plugin is busy (current operation: {})'.format(self.current_operation))
-
-    def _prelock(self):
-        # 1. Read the current operation from lockfile before timeout or direct lock call.
+        op = self.current_operation
+        if not op:
+            op = '<UNKNOWN>'
```
Comment (benjamreis) [inline SOURCES/etc/xapi.d/plugins/updater.py:37]: shouldn't `'<UNKNOWN>'` be stored in a global variable?

**Review by benjamreis — APPROVED** [review]

File: SOURCES/etc/xapi.d/plugins/updater.py (hunk 1 above)
Comment (Wescoeur) [reply SOURCES/etc/xapi.d/plugins/updater.py:37]: Why? IMHO It's useful only in this context. Maybe later if we think it's required.

Comment (benjamreis) [reply SOURCES/etc/xapi.d/plugins/updater.py:37]: Fine by me.

**Comment by benjamreis** [comment]
Please add the `pyfakefs` requirements for tests in the [README](https://github.com/xcp-ng/xcp-ng-xapi-plugins#tests).

**Review by benjamreis — APPROVED** [review]

**Review by stormi — COMMENTED** [review]
The last commit should mention that it fixes a behaviour change (the added timeout in the update - update cas) introduced by the first commit.

**Review by benjamreis — APPROVED** [review]

**Review by stormi — APPROVED** [review]

**Comment by benjamreis** [comment]
Solved: https://github.com/xcp-ng/xcp/issues/523

**Pull request merged**